### PR TITLE
Script to Parse All Tops on Reformation and upload to our database

### PR DESCRIPTION
**Pull request by treyholt**
**Problem**
In order to suggest alternatives, we need to have a catalogue of good alternatives handy in our database. So it became apparent that we would need to fill our db with as many solid alternatives before we can do any meaningful recommendation. 
One problem is that these large sustainable clothing sites don't seem to have easily accessible APIs. 

**Solution**
The solution was a bit of in-the-weeds web scraping. 
Please note this was not a generic solution, and this specific script only works for 'Tops' at reformation.com. 

The strategy is pretty simple, in a python script we want to launch the reformation page where they display their tops, scrape the exact right tags of html to pull the correct info, populate our own custom data objects with useful info (i.e price, name, color, image url etc) and then put that data onto our own database for use at a later time. 

**Hurdles**
One massive hurdle I was worried about is that of course when you load a page of items it only shows the first first 20 or so items on that page. And worse, it appeared at first that reformation wasn't actually changing their URL. It took a lot of digging around in Chrome inspect to realize that in fact there were requests made with a url containing queries, but it was simply not showing up in the address bar. Once I realized this it was a matter of looping from 1 to 30 (there were only 26 pages with data on them), then loading the correct page using the beautiful soup library in python, parsing the html (knew what to parse by analyzing the html structure in Chrome inspect), formulating the data object and posting to our database. 

**Notes**

The authentication mechanism to use firebase in this script with python uses a config file that is stored securely on my local machine. I need to confirm that we can make this Repo private before adding it to the repo. 

It also is probably just a good SWE thing to know, but what are we supposed to do with highly important files that are needed across the team for us all to have access to the same stuff. 

Also we now have every single Top sold on Reformation in our database. So that's pretty cool.

### CODE REVIEW

File: Bulk Scraping/ReformationScrape.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# When running this you'll have to download the service account info, not sure 
+# how to do this well with version control. 
+cred = credentials.Certificate('/Users/trey/Documents/suzie-1234-firebase-adminsdk-osp8h-1e8460abbb.json')
+firebase_admin.initialize_app(cred)
+
+db = firestore.client()
+URL = 'https://www.thereformation.com/categories/tops'
+
+for page_no in range(1, 30):
```
Comment (sanjayesn) [inline Bulk Scraping/ReformationScrape.py:16]: Is 30 the max number of pages we can have?

File: Bulk Scraping/ReformationScrape.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    page_no_str = str(page_no)
+    full_url = URL + "?page=" + page_no_str
+    page = requests.get(full_url)
+    soup = BeautifulSoup(page.text, 'html.parser')
+
+    print('PAGE', page_no)
+    results = soup.find_all("div", "product-summary")
+    for item in results:
+        #Navigating down to the image source
+        image_url = item.div.a.img.attrs['data-product-summary-hover-image']
```
Comment (sanjayesn) [inline Bulk Scraping/ReformationScrape.py:26]: Awesome how we can get images - might be an opportunity for a stretch goal here.

File: Bulk Scraping/ReformationScrape.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import json
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+import requests
+from bs4 import BeautifulSoup
+
+# When running this you'll have to download the service account info, not sure 
+# how to do this well with version control. 
+cred = credentials.Certificate('/Users/trey/Documents/suzie-1234-firebase-adminsdk-osp8h-1e8460abbb.json')
```
Comment (sanjayesn) [inline Bulk Scraping/ReformationScrape.py:10]: There might be a way to store encrypted secrets on GitHub, but yes - definitely don't want to store private auth keys in the open.

**Review by sanjayesn — APPROVED** [review]
Looks good. I was able to replicate your results in the sustainable_alternatives/brands/reformation_test collection on Firestore. We should look into credential management solutions on GitHub.

File: Bulk Scraping/ReformationScrape.py (hunk 1 above)
Comment (treyholt) [reply Bulk Scraping/ReformationScrape.py:16]: Nope, to be precise I should have done 26, as that is the last page with data on it, could've gone way higher if need be.

**Pull request merged**